File: src/test/java/net/sjrx/intellij/plugins/systemdunitfiles/commenter/UnitFileCommenterTest.java
package net.sjrx.intellij.plugins.systemdunitfiles.commenter;

import com.intellij.codeInsight.generation.actions.CommentByLineCommentAction;
import com.intellij.testFramework.fixtures.BasePlatformTestCase;
import net.sjrx.intellij.plugins.systemdunitfiles.filetypes.ServiceFileType;

public class UnitFileCommenterTest extends BasePlatformTestCase {

  public void testCommenter() {
    myFixture.configureByText(ServiceFileType.INSTANCE, "<caret>[Service]");
    CommentByLineCommentAction commentAction = new CommentByLineCommentAction();
    commentAction.actionPerformedImpl(getProject(), myFixture.getEditor());
    myFixture.checkResult("#[Service]");
    commentAction.actionPerformedImpl(getProject(), myFixture.getEditor());
    myFixture.checkResult("[Service]");
  }


}

File: src/test/kotlin/net/sjrx/intellij/plugins/systemdunitfiles/commenter/UnitFileCommenterTest.kt
package net.sjrx.intellij.plugins.systemdunitfiles.commenter

import com.intellij.codeInsight.generation.actions.CommentByLineCommentAction
import com.intellij.testFramework.fixtures.BasePlatformTestCase
import net.sjrx.intellij.plugins.systemdunitfiles.filetypes.ServiceFileType

class UnitFileCommenterTest : BasePlatformTestCase() {
  fun testCommenter() {
    myFixture.configureByText(ServiceFileType.INSTANCE, "<caret>[Service]")
    val commentAction = CommentByLineCommentAction()
    commentAction.actionPerformedImpl(project, myFixture.editor)
    myFixture.checkResult("#[Service]")
    commentAction.actionPerformedImpl(project, myFixture.editor)
    myFixture.checkResult("[Service]")
  }
}
